railties/lib/rails/generators/rails/app/templates/config/boot.rb.tt

```diff
@@ -1,12 +1,12 @@
-<% unless options.skip_spring? -%>
-begin
-  load File.expand_path("../bin/spring", __dir__)
-rescue LoadError => e
-  raise unless e.path.end_with?("/bin/spring")
-end
-<% end -%>
```

Comment:
I initially created my application with Rails@6.0 but recently have been running off Rails master.

I stumbled upon the Rails upgrade tool to upgrade my application from 6.0 to 6.1. Running the upgrade task caused this snippet code to be added to `boot.rb`.

However, this change prevented my application from deploying on Heroku. It appears that this code still looked for Spring although it wasn't installed as Spring is located in the `development` group within my Gemfile. I'm not sure why this is, but removing this snipped caused my application to successfully build.
```
-----> Detecting rake tasks
 !
 !     Could not detect rake tasks
 !     ensure you can run `$ bundle exec rake -P` against your app
 !     and using the production group of your Gemfile.
 !     /tmp/build_acae070e_/config/boot.rb:4:in `rescue in <top (required)>': undefined method `end_with?' for nil:NilClass (NoMethodError)
 !     from /tmp/build_acae070e_/config/boot.rb:1:in `<top (required)>'
 !     from /tmp/build_acae070e_/bin/rake:7:in `require_relative'
 !     from /tmp/build_acae070e_/bin/rake:7:in `<main>'
 !     /tmp/build_acae070e_/vendor/ruby-2.7.1/lib/ruby/2.7.0/rubygems/dependency.rb:311:in `to_specs': Could not find 'spring' (= 2.1.1) among 140 total gem(s) (Gem::MissingSpecError)
 !     Checked in 'GEM_PATH=/tmp/build_acae070e_/vendor/bundle/ruby/2.7.0', execute `gem env` for more information
 !     from /tmp/build_acae070e_/vendor/ruby-2.7.1/lib/ruby/2.7.0/rubygems/dependency.rb:323:in `to_spec'
 !     from /tmp/build_acae070e_/vendor/ruby-2.7.1/lib/ruby/2.7.0/rubygems/core_ext/kernel_gem.rb:62:in `gem'
 !     from /tmp/build_acae070e_/bin/spring:14:in `<top (required)>'
 !     from /tmp/build_acae070e_/config/boot.rb:2:in `load'
 !     from /tmp/build_acae070e_/config/boot.rb:2:in `<top (required)>'
 !     from /tmp/build_acae070e_/bin/rake:7:in `require_relative'
 !     from /tmp/build_acae070e_/bin/rake:7:in `<main>'
```